hdmf-dev/hdmf
# Fix internal links in docs

## Motivation

See related PR in https://github.com/NeurodataWithoutBorders/pynwb/pull/1827

 There are some broken internal links in the hdmf documentation that impact pynwb as well.
- Links to h5py/pandas data types are not currently working - (see [example in parameters section here](https://hdmf.readthedocs.io/en/stable/hdmf.query.html#hdmf.query.HDMFDataset)) 
- Links to return types are not generated if a function has more than one return type, or a nested return type (see [example in the get_container method here](https://hdmf.readthedocs.io/en/stable/hdmf.common.multi.html#hdmf.common.multi.SimpleMultiContainer.get_container))

## How to test the behavior?
To see linkcheck warnings:
```
sphinx-build -n -b linkcheck ./docs/source ./docs/_build
```

To see updated docs:
```
cd docs
make html
```

## Checklist
